jax/_src/scipy/special.py

```diff
@@ -724,82 +725,249 @@ def i1(x: ArrayLike) -> Array:
-                false_fn_update_bs, operand=(bs, f))
-
-  f0 = f1
-  f1 = f
-  return (f0, f1, bs, z), f
-
-
-def _bessel_jn(z: ArrayLike, *, v: int, n_iter: int=50) -> Array:
-  f0 = _lax_const(z, 0.0)
-  f1 = _lax_const(z, 1E-16)
-  f = _lax_const(z, 0.0)
-  bs = _lax_const(z, 0.0)
-
-  (_, _, bs, _), j_vals = lax.scan(
-      f=_bessel_jn_scan_body_fun, init=(f0, f1, bs, z),
-      xs=lax.iota(lax.dtype(z), n_iter+1), reverse=True)
-
-  f = j_vals[0]  # Use the value at the last iteration.
-  j_vals = j_vals[:v+1]
-  j_vals = j_vals / (bs - f)
-
-  return j_vals
-
+# polynomial coefficients for J1
+
+RP1 = [-8.99971225705559398224E8, 4.52228297998194034323E11,
+-7.27494245221818276015E13, 3.68295732863852883286E15,]
+RQ1 = [1.0, 6.20836478118054335476E2, 2.56987256757748830383E5, 8.35146791431949253037E7, 
+ 2.21511595479792499675E10, 4.74914122079991414898E12, 7.84369607876235854894E14, 
+ 8.95222336184627338078E16, 5.32278620332680085395E18,]
+
+PP1 = [7.62125616208173112003E-4, 7.31397056940917570436E-2, 1.12719608129684925192E0, 
+ 5.11207951146807644818E0, 8.42404590141772420927E0, 5.21451598682361504063E0, 1.00000000000000000254E0,]
+PQ1 = [5.71323128072548699714E-4, 6.88455908754495404082E-2, 1.10514232634061696926E0, 
+ 5.07386386128601488557E0, 8.39985554327604159757E0, 5.20982848682361821619E0, 9.99999999999999997461E-1,]
+
+QP1 = [5.10862594750176621635E-2, 4.98213872951233449420E0, 7.58238284132545283818E1, 
+ 3.66779609360150777800E2, 7.10856304998926107277E2, 5.97489612400613639965E2, 2.11688757100572135698E2, 2.52070205858023719784E1,]
+QQ1  = [1.0, 7.42373277035675149943E1, 1.05644886038262816351E3, 4.98641058337653607651E3, 
+ 9.56231892404756170795E3, 7.99704160447350683650E3, 2.82619278517639096600E3, 3.36093607810698293419E2,]
+
+YP1 = [1.26320474790178026440E9,-6.47355876379160291031E11, 1.14509511541823727583E14,
+ -8.12770255501325109621E15, 2.02439475713594898196E17,-7.78877196265950026825E17,]
+YQ1 = [5.94301592346128195359E2, 2.35564092943068577943E5, 7.34811944459721705660E7, 
+ 1.87601316108706159478E10, 3.88231277496238566008E12, 6.20557727146953693363E14, 
+ 6.87141087355300489866E16, 3.97270608116560655612E18,]
+
+Z1 = 1.46819706421238932572E1
+Z2 = 4.92184563216946036703E1
+PIO4 = .78539816339744830962 # pi/4
+THPIO4 = 2.35619449019234492885 # 3*pi/4
+SQ2OPI = .79788456080286535588 # sqrt(2/pi)
+
+# polynomial coefficients for J0
+
+PP0 = [7.96936729297347051624E-4,
+  8.28352392107440799803E-2,
+  1.23953371646414299388E0,
+  5.44725003058768775090E0,
+  8.74716500199817011941E0,
+  5.30324038235394892183E0,
+  9.99999999999999997821E-1,
+]
+PQ0 = [
+  9.24408810558863637013E-4,
+  8.56288474354474431428E-2,
+  1.25352743901058953537E0,
+  5.47097740330417105182E0,
+  8.76190883237069594232E0,
+  5.30605288235394617618E0,
+  1.00000000000000000218E0,
+]
 
-@partial(jit, static_argnames=["v", "n_iter"])
-def bessel_jn(z: ArrayLike, *, v: int, n_iter: int=50) -> Array:
```

Comment:
Let's leave the existing `bessel_jn` function, and plan to deprecate it as a followup to this PR.

Reply:
Ok! It's pretty bad though.

Reply:
Committed this back in.

Reply:
Agreed, but `bessel_jn` is a public API and so we need to conform to the [API compatibility policy](https://jax.readthedocs.io/en/latest/api_compatibility.html). I can take care of the deprecation in a followup once this change lands.